src/lib/ui/AccessCodeForm/AccessCodeForm.tsx

```diff
@@ -206,81 +205,83 @@ function Content({
               }}
             />
           </FormField>
-          {hasCodeInputs && (
-            <>
-              <FormField className='seam-code-field'>
-                <InputLabel>{t.codeInputLabel}</InputLabel>
-                <TextField
-                  size='large'
-                  clearable
-                  hasError={hasCodeError}
-                  helperText={responseErrors?.code ?? errors.code?.message}
-                  inputProps={{
-                    ...register('code', {
-                      required: t.codeRequiredError,
-                      validate: (value: string) =>
-                        validateCodeLength(device, value),
-                    }),
-                  }}
-                />
-                <div
-                  className={classNames('seam-bottom', {
-                    'has-hints': codeLengthRequirementMessage != null,
-                  })}
-                >
-                  {codeLengthRequirementMessage != null && (
-                    <ul
-                      className={classNames('seam-requirements', {
-                        'seam-error': hasCodeError,
-                      })}
-                    >
-                      <li>{codeLengthRequirementMessage}</li>
-                      <li>{t.codeNumbersOnlyRequirement}</li>
-                    </ul>
-                  )}
-                  <Button
-                    size='small'
-                    onMouseDown={(e) => {
-                      e.preventDefault() // Disable stealing input focus
-                      handleGenerateCode()
+          {hasCodeInputs !== null &&
+            hasCodeInputs !== undefined &&
+            hasCodeInputs && (
```

Comment:
```suggestion
          {hasCodeInputs && (
```

Reply:
Yes, it's redundant, I have removed it now